AprilSylph/XKit-Rewritten
# Tag Tracking+: Sidebar filtering option; no-tags placeholder

### Description


This adds an option to only show tags with new posts in the sidebar, and adds status-indicating placeholder elements to the sidebar for a) not having any tracked tags, b) having tracked tags, all of which are hidden by the option, and c) having tracked tags which are still loading their counts and are hidden by the option.

I think they should probably look different than regular sidebar element list items, but I have no ideas for the CSS. Head empty.

This code is kind of messy since we're manipulating the utility-generated sidebar DOM elements. I keep edging into model-view-controller type structures ("make it look like react") and backing off.

resolves #832, resolves #833

### Testing steps


- Enable the sidebar.
- With the new "only show tags with new posts in the sidebar" option disabled, follow 0 tags. There should be a status placeholder in the sidebar.
- Follow some tags and refresh. There should be no status placeholder, all of the tags should appear, and they should update their counts in sequence. Opening the search dropdown mid-update should cause no problems.
- With the new "only show tags with new posts in the sidebar" option enabled, track some tags with new posts (or add `if (Math.random() < 0.3) unreadCount = 5;` at line 56.) There should be a loading placeholder in the sidebar, then some tags with counts. Opening the search dropdown mid-update should cause no problems.
- Clear the new posts. There should now be a status placeholder in the sidebar. Hard refresh. There should be a loading placeholder in the sidebar, then a a status placeholder.
- Open a page with no sidebar (https://www.tumblr.com/april) and soft navigate to the dashboard; no problems should occur.
